### PR TITLE
fix mistake in the heapsort() method and add MaxPriorityQueue.java

### PR DESCRIPTION
**Pull request by yuhsin7676**
### **Describe your change:**

Fixes https://github.com/TheAlgorithms/Java/issues/3515

- Fix a bug in the algorithm heapsort() in the MinPriorityQueue.java of com.thealgorithms.datastructures.heaps package. heapsort() method gave an error when size < capacity. Also add MaxPriorityQueue.

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by debasishbsws — COMMENTED** [review]
I think it would be better to use `ArrayList` instead of `int` so the capacity becomes dynamic. and you can also make these using generics, to fit every type of data structure not only int.

**Review by siriak — CHANGES_REQUESTED** [review]
Please replace `main()`s with JUnit tests

**Comment by yuhsin7676** [comment]
> Please replace `main()`s with JUnit tests

OK

**Comment by debasishbsws** [comment]
I think there are many files that don't have any dedicated test files, rather just a `main()` inside the class.

**Comment by siriak** [comment]
@debasishbsws that'd be nice!

File: src/test/java/com/thealgorithms/datastructures/lists/CursorLinkedListTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    public static void tearDownClass() {
+    }
+    
+    @BeforeEach
+    public void setUp() {
+    }
+    
+    @AfterEach
+    public void tearDown() {
+    }
```
Comment (siriak) [inline src/test/java/com/thealgorithms/datastructures/lists/CursorLinkedListTest.java:37]: This is redundant

File: src/test/java/com/thealgorithms/datastructures/lists/CursorLinkedListTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
+/*
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
+ * Click nbfs://nbhost/SystemFileSystem/Templates/UnitTests/JUnit5TestClass.java to edit this template
+ */
```
Comment (siriak) [inline src/test/java/com/thealgorithms/datastructures/lists/CursorLinkedListTest.java:4]: What is that?

File: nb-configuration.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <properties xmlns="http://www.netbeans.org/ns/maven-properties-data/1">
+        <!--
+Properties that influence various parts of the IDE, especially code formatting and the like. 
+You can copy and paste the single properties, into the pom.xml file and the IDE will pick them up.
+That way multiple projects can share the same settings (useful for formatting rules for example).
+Any value defined here will override the pom.xml file value but is only applicable to the current project.
+-->
+        <netbeans.hint.jdkPlatform>JDK_8</netbeans.hint.jdkPlatform>
+    </properties>
+</project-shared-configuration>
```
Comment (siriak) [inline nb-configuration.xml:18]: Please remove this file

File: pom.xml [hunk 4]
```diff
@@ -58,8 +58,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.10.1</version>
                 <configuration>
-                    <source>17</source>
-                    <target>17</target>
+                    <source>1.8</source>
+                    <target>1.8</target>
```
Comment (siriak) [inline pom.xml:62]: Revert these changes

**Review by siriak — CHANGES_REQUESTED** [review]

**Pull request closed without merging**